frontend/public/actions/k8s.ts

```diff
@@ -79,217 +45,11 @@ export const getResources = () => (dispatch: Dispatch) => {
-   *   2. the WS can not establish a connection within $TIMEOUT
-   */
-  const pollAndWatch = async () => {
-    delete POLLs[id];
-
-    try {
-      const resourceVersion = await incrementallyLoad();
-      // ensure this watch should still exist because pollAndWatch is recursiveish
-      if (!REF_COUNTS[id]) {
-        // eslint-disable-next-line no-console
-        console.log(`stopped watching ${id} before finishing incremental loading.`);
-        // call cleanup function out of abundance of caution...
-        dispatch(stopK8sWatch(id));
-        return;
-      }
-
-      if (WS[id]) {
-        // eslint-disable-next-line no-console
-        console.warn(`Attempted to create multiple websockets for ${id}.`);
-        return;
-      }
-
-      if (!_.get(k8skind, 'verbs', ['watch']).includes('watch')) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `${referenceForModel(k8skind)} does not support watching, falling back to polling.`,
-        );
-        if (!POLLs[id]) {
-          POLLs[id] = setTimeout(pollAndWatch, 15 * 1000);
-        }
-        return;
-      }
-
-      const { subprotocols } = getImpersonate(getState()) || {};
-      WS[id] = k8sWatch(
-        k8skind,
-        { ...query, resourceVersion },
-        { subprotocols, timeout: 60 * 1000 },
-      );
-    } catch (e) {
-      if (!REF_COUNTS[id]) {
-        // eslint-disable-next-line no-console
-        console.log(`stopped watching ${id} before finishing incremental loading with error ${e}!`);
-        // call cleanup function out of abundance of caution...
-        dispatch(stopK8sWatch(id));
-        return;
-      }
-
-      dispatch(errored(id, e));
-
-      if (!POLLs[id]) {
-        POLLs[id] = setTimeout(pollAndWatch, 15 * 1000);
-      }
-      return;
-    }
-
-    WS[id]
-      .onclose((event) => {
-        // Close Frame Status Codes: https://tools.ietf.org/html/rfc6455#section-7.4.1
-        if (event.code !== 1006) {
-          return;
-        }
-        // eslint-disable-next-line no-console
-        console.log('WS closed abnormally - starting polling loop over!');
-        const ws = WS[id];
-        ws && ws.destroy();
-      })
-      .ondestroy((timedOut) => {
-        if (!timedOut) {
-          return;
-        }
-        // If the WS is unsucessful for timeout duration, assume it is less work
-        //  to update the entire list and then start the WS again
-
-        // eslint-disable-next-line no-console
-        console.log(`${id} timed out - restarting polling`);
-        delete WS[id];
-
-        if (POLLs[id]) {
-          return;
-        }
-
-        POLLs[id] = setTimeout(pollAndWatch, 15 * 1000);
-      })
-      .onbulkmessage((events) =>
-        [updateListFromWS, extraAction].forEach((f) => f && dispatch(f(id, events))),
-      );
-  };
-  pollAndWatch();
-};
+export {
+  watchK8sObject,
+  watchK8sList,
+  stopK8sWatch,
+} from '@console/dynamic-plugin-sdk/src/app/k8s/actions/k8s';
```

Comment:
Nit: We should really be exporting at the top of the file (under the imports) or bottom of the file, not nesting in the middle.